internal/repository/sync.go

```diff
@@ -64,8 +68,49 @@ func Clone(ctx context.Context, remoteURL, dir, initialBranch string) (*Reposito
 		return nil, errors.Join(ErrCloningRepository, err)
 	}
 
+	sort.Slice(expectedRootKeys, func(i, j int) bool {
+		return expectedRootKeys[i].KeyID < expectedRootKeys[j].KeyID
+	})
+
 	repository := &Repository{r: r}
 
 	slog.Debug("Verifying HEAD...")
+	if len(expectedRootKeys) > 0 {
+		slog.Debug("Getting root keys...")
+		state, err := policy.LoadCurrentState(ctx, r)
+		if err != nil {
+			return repository, errors.Join(ErrCloningRepository, err)
+		}
+
+		rootMetadata, err := state.GetRootMetadata()
+		if err != nil {
+			return repository, errors.Join(ErrCloningRepository, err)
+		}
+
+		if len(rootMetadata.Roles[policy.RootRoleName].KeyIDs) != len(expectedRootKeys) {
+			expectedRootKeysIDs := []string{}
+
+			for _, key := range expectedRootKeys {
+				expectedRootKeysIDs = append(expectedRootKeysIDs, key.KeyID)
+			}
+
+			err := fmt.Errorf("cloned root keys do not match the expected keys.\n Expected keys = %s, \n Actual Keys = %s", expectedRootKeysIDs, rootMetadata.Roles[policy.RootRoleName].KeyIDs)
+			return repository, errors.Join(ErrCloningRepository, err)
+		}
+
+		slog.Debug("Verifing if root keys are expected root keys...")
+		for keyIndex := range expectedRootKeys {
+			expectedRootKeysIDs := []string{}
+
+			for _, key := range expectedRootKeys {
+				expectedRootKeysIDs = append(expectedRootKeysIDs, key.KeyID)
+			}
+
+			if !reflect.DeepEqual(rootMetadata.Keys[rootMetadata.Roles[policy.RootRoleName].KeyIDs[keyIndex]], expectedRootKeys[keyIndex]) {
+				err := fmt.Errorf("cloned root keys do not match the expected keys.\n Expected keys = %s, \n Actual Keys = %s", expectedRootKeysIDs, rootMetadata.Roles[policy.RootRoleName].KeyIDs)
```

Comment:
DRY with above, perhaps this can be a named error / string that's formatted before being wrapped in an error.

Reply:
Fixed